### PR TITLE
BN-973 Insert canonical head vertex, partial Pr

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
update canonical head vertex during Genus insert block task 

## Approach

- Canonical head vertex, was implemented in a previous or #2913 
- This PR, update this vertex (delete and create a new vertex) with a link to the Block header head of the chain. (no edges was created, just a link)
- When genus server start, get the latest canonical head changes from the Node, and then fetch the head on the graph.
- With this information, the two heights, we fetch blocks the missing blocks and insert them into the graph.

## Testing
Start an stop multiple times the Genus app, and validate that we only have 1 canonical head vertex, liked to the head

## Tickets
Partial Pr BN-973
Screenshot from 2023-04-19 13-32-54

### CODE REVIEW

File: genus-server/src/main/resources/environment.conf [hunk 1]
```diff
@@ -9,7 +9,8 @@ genus {
   // The _base_ directory in which genus_db data is stored
   // optionally if OrientDB Studio was previously installed required, the genus_db storage should be a child of ¨database¨ folder
   // Example = "../../orientdb-community-3.2.18/databases/genus_db"
-  orient-db-directory = "/tmp/genus_db"
+  orient-db-directory = "../../orientdb-community-3.2.18/databases/genus_db"
```
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply genus-server/src/main/resources/environment.conf:12]: accident, thanks for the catch

File: genus-server/src/main/scala/co/topl/genusServer/GenusServerApp.scala [hunk 2]
```diff
@@ -41,20 +44,40 @@ object GenusServerApp
...
+          }
             .evalTap(res => Logger[F].info(res.leftMap(_.toString).swap.getOrElse("OK")))
         )
         .toResource
 
       _ <-
         if (nodeEnabled)
           inserter
-            .take(300)
+            .take(Integer.MAX_VALUE)
```
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply genus-server/src/main/scala/co/topl/genusServer/GenusServerApp.scala:80]: good catch, my mistake

File: genus-library/src/main/scala/co/topl/genusLibrary/orientDb/instances/VertexSchemaInstances.scala [hunk 3]
```diff
@@ -32,6 +32,16 @@ object VertexSchemaInstances {
 
       def addIoTx(ioTx: IoTransaction): OrientVertex =
         graph.addVertex(s"class:${ioTransactionSchema.name}", ioTransactionSchema.encode(ioTx).asJava)
+
+      def addCanonicalHead(blockHeaderVertex: OrientVertex): OrientVertex = {
+        // Is expected that Canonical head schema only contains 1 vertex, the head of the chain
+        graph.getVerticesOfClass(s"${canonicalHeadSchema.name}").asScala.foreach(graph.removeVertex)
+        val v = graph
+          .addVertex(s"class:${canonicalHeadSchema.name}", canonicalHeadSchema.encode(CanonicalHead).asJava)
```
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply genus-library/src/main/scala/co/topl/genusLibrary/orientDb/instances/VertexSchemaInstances.scala:40]: there is no update operation, but I will update it with another approach and maybe it is more clear.

File: genus-server/src/main/scala/co/topl/genusServer/GenusServerApp.scala [hunk 4]
```diff
@@ -41,20 +44,40 @@ object GenusServerApp
       nodeBlockFetcher <- NodeBlockFetcher.make(rpcInterpreter)
 
       // TODO this is just proof of concept, we need to add a lot of logic here, related to retries and handling errors
-      nodeEnabled <- Resource.pure(false) // maybe we can use a config
+      nodeEnabled <- Resource.pure(true) // maybe we can use a config
+
+      nodeLatestHeight <- nodeBlockFetcher
+        .fetchHeight()
```
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply genus-server/src/main/scala/co/topl/genusServer/GenusServerApp.scala:50]: thanks for the tip!

**Review by SeanCheatham — APPROVED** [review]

**Pull request merged**